CAAppKitBridge/Source/NSView+CAMethods.m

```diff
@@ -120,56 +120,21 @@ - (CALayer *) makeBackingLayer
   return [CALayer layer];
 }
 
-
-/* methods from libs-gui/Headers/AppKit/NSOpenGLView.h */
-
-
-/**
- detach from the current context.  You should call it before releasing this 
- object.
-*/
-- (void) clearGLContext
-{
-  if (self->_coreAnimationData == nil) {
-      return;
-  }
-  GSCAData *currGSCAData = self->_coreAnimationData;
-  NSOpenGLContext *currGlContext = currGSCAData->_GLContext;
-if (currGlContext)
-  {
-    [currGlContext clearDrawable];
-    DESTROY(currGlContext);
-    currGSCAData->_prepared = NO;
-  }
-}
-
-- (void) setOpenGLContext: (NSOpenGLContext*)context
-{
-  GSCAData *currGSCAData = self->_coreAnimationData;
-  NSOpenGLContext *currGlContext = currGSCAData->_GLContext;
-  if ( context != currGlContext )
-      {
-      [self clearGLContext];
-      ASSIGN(currGlContext, context);
-      }
-}
-
-- (NSOpenGLContext*) openGLContext
+- (NSOpenGLContext*) _gsCreateOpenGLContext
 {
   GSCAData *currGSCAData = self->_coreAnimationData;
-  NSOpenGLContext *currGlContext = currGSCAData->_GLContext;
-  if (currGlContext == nil)
-  {
-    NSOpenGLContext *context = [[NSOpenGLContext alloc] 
-                                   initWithFormat: currGSCAData->_pixelFormat
-                                   shareContext: nil];
-
-    [self setOpenGLContext: context];
-    [context setView: self];
+  NSOpenGLContext *currGLContext = currGSCAData->_GLContext;
+  if (currGLContext == nil)
+    {
+      NSOpenGLContext *context = [[NSOpenGLContext alloc] 
+                                     initWithFormat: currGSCAData->_pixelFormat
+                                     shareContext: nil];
 
-    RELEASE(context);
-  }
-  return currGlContext;
+      ASSIGN(currGLContext, context);
+      [context setView: self];
+      RELEASE(context);
+    }
+  return currGLContext;
```

Comment:
This is going to return an invalid, `dealloc`'d pointer because the pointer is released. At least the pointer is not being stored for future (and you are every time allocating a new context), but you are still returning an invalid pointer.